src/browser/Terminal.ts

```diff
@@ -71,12 +74,18 @@ export class Terminal extends CoreTerminal implements ITerminal {
   private _helperContainer: HTMLElement | undefined;
   private _compositionView: HTMLElement | undefined;
 
+  private _overviewRulerRenderer: OverviewRulerRenderer | undefined;
+
   // private _visualBellTimer: number;
 
   public browser: IBrowser = Browser as any;
 
   private _customKeyEventHandler: CustomKeyEventHandler | undefined;
 
+  // TODO: Move into CoreTerminal.ts
+  // common services
+  private _decorationService: DecorationService;
+
   // browser services
   private _charSizeService: ICharSizeService | undefined;
   private _mouseService: IMouseService | undefined;
```

Comment:
Actually, I changed my mind about this overnight. Since we're not supporting decorations in xterm-headless at all, this should live in the `browser/` file so code that never runs doesn't end up showing up in the headless bundle. I'm a little on the fence about whether `DecorationService` should live in common or browser, let's keep it in common for now since it won't be included in the compiled bundle unless it's referenced.

```suggestion
  // browser services
  private _decorationService: DecorationService;
  private _charSizeService: ICharSizeService | undefined;
  private _mouseService: IMouseService | undefined;
```